nerfstudio/engine/trainer.py

```diff
@@ -503,3 +506,11 @@ def eval_iteration(self, step: int) -> None:
         if step_check(step, self.config.steps_per_eval_all_images):
             metrics_dict = self.pipeline.get_average_eval_image_metrics(step=step)
             writer.put_dict(name="Eval Images Metrics Dict (all images)", scalar_dict=metrics_dict, step=step)
+
+    def on_train_end(self) -> None:
+        """Calls optional on_train_end() model method at training end"""
+        if callable(self.pipeline.model.on_train_end()):
+            try:
```

Comment:
This is bad, it hides the exception. Why did you add the try-catch in the first place?

Reply:
I put the try-catch in case a user implements something that fails within the function, so it wouldn't break the trainer.